internal/datasources/toml.go

```diff
@@ -22,10 +23,22 @@ func (ds *TomlDatasource) Load() (map[string]any, error) {
 	defer file.Close()
 
 	data := make(map[string]any)
-	decoder := toml.NewDecoder(file)
-	if err := decoder.Decode(&data); err != nil {
+	data, err = DecodeToml(file, data)
```

Comment:
Perhaps we can change the yaml,toml,json data sources to work with a reader instead of a file?
This way we can use a single Load function for both files and http